question.php

```diff
@@ -144,14 +165,30 @@ public function apply_attempt_state(question_attempt_step $step) {
 
         /* TODO: This method is also called from question_attempt->regrade and
                  question_attempt->start_question_based_on, where we shouldn't need to get the UI. */
-        $attempt = $this->api->package($this->packagehash, $this->packagefile)
-            ->view_attempt(
-                $this->questionstate,
-                $this->attemptstate,
-                $this->scoringstate,
-                $lastresponse
-            );
-        $this->update_ui($attempt->ui);
+        try {
+            $attempt = $this->api->package($this->packagehash, $this->packagefile)
+                ->view_attempt(
+                    $this->questionstate,
+                    $this->attemptstate,
+                    $this->scoringstate,
+                    $lastresponse
+                );
+            $this->update_ui($attempt->ui);
+        } catch (Throwable $t) {
+            // Trigger error event.
+            // TODO: get question_attempt-id via DB-query (question_attempt_step-id -> question_attempt-id)?
+            $params = [
+                'context' => $PAGE->context,
+                'relateduserid' => $step->get_user_id(),
+                'other' => [
+                    'questionid' => $this->id,
+                    'errormessage' => $t->getMessage(),
+                ],
+            ];
+            $event = \qtype_questionpy\event\viewing_attempt_failed::create($params);
+            $event->trigger();
+            debugging($event->get_description());
```

Comment:
Wenn ich das richtig sehe, ist `$this->get_behaviour()->get_qa()->get_database_id()` hier dann immer gesetzt.